Dockerfile

```diff
@@ -53,8 +53,8 @@ RUN apk add --update musl go=$GO_VERSION git bash wget ca-certificates \
 	# We get the current commit using this hack,
 	# so that we don't have to copy all of .git/ into the build context.
 	# This saves us quite a bit of image size.
-	&& ref="$(cat .git/HEAD | cut -d' ' -f2)" \
-	&& commit="$(cat .git/$ref | head -c 7)" \
+	&& ref=$(cat .git/HEAD | grep ref | cut -d' ' -f2) \
+	&& commit=$(if [ -z "$ref" ]; then cat .git/HEAD; else cat ".git/$ref"; fi | head -c 7) \
```

Comment:
You need the following for `rev-parse` and `symbolic-ref` to work:

- `.git/HEAD`
- `.git/refs/`
- `.git/objects/` which can be empty

See: https://github.com/git/git/blob/master/setup.c#L266

So it might be a good idea to just `mkdir .git/objects/` so that you can use git commands afterwards.

It could be also a good idea to copy `.git/packed-refs` if it exists in case the refs are packed.